### PR TITLE
[SYCL-MLIR]: Add cmake target to run Polygeist unittests

### PR DESCRIPTION
**Pull request by etiotto**
Reorganized the cmake files for Polygeist tests to allow running the Polygeist lit test and unittests via:
-  ninja check-polygeist (to run the lit tests)
-  ninja check-polygeist-unit (to run the unittests)

This make Polygeist more consistent with the cmake structure of MLIR.

### CODE REVIEW

**Comment by etiotto** [comment]
Depends on #9292

File: .github/workflows/sycl_linux_build_and_test.yml [hunk 1]
```diff
@@ -111,6 +111,7 @@ jobs:
       run: cmake --build $GITHUB_WORKSPACE/build --target deploy-sycl-toolchain
     # TODO allow to optionally disable in-tree checks
     - name: check-mlir-sycl
+      shell: bash
```
Comment (victor-eds) [inline .github/workflows/sycl_linux_build_and_test.yml:114]: Why do we need this change? Is this not working somehow now?

File: polygeist/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,4 @@
-# Copyright (C) Codeplay Software Limited
-
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.13.4)
```
[the remark this answers is not in the document]
Comment (etiotto) [reply polygeist/CMakeLists.txt:1]: I bumped it up to match the version number MLIR itself uses.

[the remark this answers is not in the document]
Comment (victor-eds) [reply polygeist/CMakeLists.txt:1]: LGTM. Not sure if this fits this PR, though.

File: polygeist/test/Unit/lit.site.cfg.py.in [hunk 3]
```diff
@@ -0,0 +1,10 @@
+@LIT_SITE_CFG_IN_HEADER@
+
+import sys
+
+config.llvm_tools_dir = lit_config.substitute("@LLVM_TOOLS_DIR@")
+config.llvm_build_mode = lit_config.substitute("@LLVM_BUILD_MODE@")
+config.polygeist_obj_root = "@POLYGEIST_BINARY_DIR@"
+
```
Comment (whitneywhtsang) [inline polygeist/test/Unit/lit.site.cfg.py.in:8]: Do we need the lines below?
```
import lit.llvm
lit.llvm.initialize(lit_config, config)
```

File: polygeist/test/CMakeLists.txt [hunk 4]
```diff
@@ -6,13 +6,35 @@ configure_lit_site_cfg(
...
-    DEPENDS 
-      polygeist-opt
-      FileCheck count not
+    DEPENDS ${MLIR_POLYGEIST_TEST_DEPENDS}
     ARGS -v
 )
+set_target_properties(check-polygeist PROPERTIES FOLDER "Tests")
 
-set_target_properties(check-polygeist-opt PROPERTIES FOLDER "Tests")
+add_lit_testsuites(polygeist ${CMAKE_CURRENT_SOURCE_DIR}
```
Comment (whitneywhtsang) [inline polygeist/test/CMakeLists.txt:38]: What does this test?

File: .github/workflows/sycl_linux_build_and_test.yml [hunk 5]
```diff
@@ -122,10 +123,16 @@ jobs:
...
+      if: ${{ always() && !cancelled() && steps.build.outcome == 'success' }}
+      run: |
+        cmake --build $GITHUB_WORKSPACE/build --target check-polygeist
+    - name: check-polygeist-unit
+      shell: bash    
       if: ${{ always() && !cancelled() && steps.build.outcome == 'success' }}
       run: |
-        cmake --build $GITHUB_WORKSPACE/build --target check-polygeist-opt
+        cmake --build $GITHUB_WORKSPACE/build --target check-polygeist-unit
```
Comment (whitneywhtsang) [inline .github/workflows/sycl_linux_build_and_test.yml:135]: Where is target `check-polygeist-unit` created?

File: polygeist/test/Unit/lit.cfg.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import subprocess
+
+import lit.formats
+import lit.util
+
+# name: The name of this test suite.
+config.name = 'polygeist-Unit'
+
+# suffixes: A list of file extensions to treat as test files.
+config.suffixes = []
```
Comment (whitneywhtsang) [inline polygeist/test/Unit/lit.cfg.py:15]: Why is this an empty list?

**Comment by etiotto** [comment]
The failure in the LIT is because the script used to run the LIT tests is not the one in *this* PR, it is the committed version. 
So that is a false failure.

File: polygeist/test/Unit/lit.cfg.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+# testFormat: The test format to use to interpret tests.
+config.test_format = lit.formats.GoogleTest(config.llvm_build_mode, 'Tests')
+
+# Propagate the temp directory. Windows requires this because it uses \Windows\
+# if none of these are present.
+if 'TMP' in os.environ:
+    config.environment['TMP'] = os.environ['TMP']
+if 'TEMP' in os.environ:
+    config.environment['TEMP'] = os.environ['TEMP']
+
```
Comment (victor-eds) [inline polygeist/test/Unit/lit.cfg.py:31]: I saw the MLIR file is propagating `'HOME'`. Why don't we do it here too?

**Review by victor-eds — APPROVED** [review]

**Review by whitneywhtsang — APPROVED** [review]

File: polygeist/test/Unit/lit.cfg.py (hunk 7 above)
Comment (etiotto) [reply polygeist/test/Unit/lit.cfg.py:31]: I didn't think is needed but we can add it in a future PR if necessary

**Pull request merged**